scripts/release-note.py

```diff
@@ -52,25 +56,29 @@ def extract_release_contents(target: str, cur_tag: str, prev_tag: str) -> str:
 
     with open(target, "r") as f:
         document = f.read()
-    start_marker = f"## [{cur_tag}]"
-    start_pos = document.find(start_marker) + len(start_marker) + len(" - YYYY-MM-DD")
-    if start_pos != -1:
-        if prev_tag is None:
-            end_pos = document.find("<!-- Release links -->")
-            if end_pos == -1:
-                match = re.search(r"## \[v\d+\.\d+\.\d+] - \d{4}-\d{2}-\d{2}", document[start_pos:])
-                if match is None:
-                    end_pos = -1
-                else:
-                    end_pos = start_pos + match.start() - 1
-        else:
-            end_marker = f"## [{prev_tag}]"
-            end_pos = document.find(end_marker)
-            if end_pos == -1:
-                match = re.search(r"## \[v\d+\.\d+\.\d+] - \d{4}-\d{2}-\d{2}", document[start_pos:])
-                end_pos = start_pos + match.start() - 1
+
+    start_marker = f"## [{cur_tag}] - "
+    start_pos = document.find(start_marker)
+
+    if start_pos == -1:
+        raise ValueError(f"Start marker for tag '{cur_tag}' not found in the document.")
+
+    start_pos += len(start_marker) + len("YYYY-MM-DD")
+
+    if prev_tag:
```

Comment:
You may reduce duplication by `end_marker = f"## [{prev_tag}] if prev_tag else "<!-- Release links -->"`.